src/openshift/components/CryostatContainer.tsx

```diff
@@ -17,12 +17,55 @@ import * as React from 'react';
 import { store } from '@app/Shared/Redux/ReduxStore';
 import { Provider } from 'react-redux';
 import { CryostatController } from './CryostatController';
+import { CryostatContext, ServiceContext, Services } from '@app/Shared/Services/Services';
+import { TargetService } from '@app/Shared/Services/Target.service';
+import { SettingsService } from '@app/Shared/Services/Settings.service';
+import { LoginService } from '@app/Shared/Services/Login.service';
+import { ApiService } from '@app/Shared/Services/Api.service';
+import { NotificationsInstance } from '@app/Shared/Services/Notifications.service';
+import { NotificationChannel } from '@app/Shared/Services/NotificationChannel.service';
+import { ReportService } from '@app/Shared/Services/Report.service';
+import { TargetsService } from '@app/Shared/Services/Targets.service';
+import { pluginServices } from '@console-plugin/services/PluginContext';
+import { Observable } from 'rxjs';
+
+export const pluginContext: CryostatContext = {
+  url: (path?: string): Observable<string> => pluginServices.plugin.proxyUrl(path),
+  headers: () => {
+    const headers = new Headers({
+      // TODO populate these with context selections, maybe taken from redux or localstorage
+      'CRYOSTAT-SVC-NS': '',
+      'CRYOSTAT-SVC-NAME': '',
+    });
+    return headers;
+  },
+};
```

Comment:
This is the first part of the key change here. It overrides the default implementation that sets up all of these various frontend "service" pieces and the context object that holds a reference to them, so that all the various components at various levels of the tree can do things like make API requests or subscribe to notifications. In particular, using the extension point provided by https://github.com/cryostatio/cryostat-web/pull/1542 , we override the URL provider to one that uses the proxy URL as a base, and override the headers provider to one that can inject the custom `<namespace, name>` Service selection parameters for use by the plugin backend.

https://github.com/cryostatio/cryostat-web/blob/8797974f25872d314e633210a2d9c0ac1f6a549b/src/app/Shared/Services/Services.tsx#L44